lib/improver/tests/wind_calculations/wind_components/test_ResolveWindComponents.py

```diff
@@ -0,0 +1,230 @@
+
+    def test_basic(self):
+        """Tests the output string is as expected"""
+        result = str(ResolveWindComponents())
+        self.assertEqual(result, '<ResolveWindComponents>')
+
+
+class Test_calc_true_north_offset(IrisTest):
+    """Tests the calc_true_north_offset function"""
+
+    def setUp(self):
+        """Set up a target cube with OSGB projection"""
+        wind_angle = np.zeros((3, 5), dtype=np.float32)
+        self.directions = set_up_cube(
+            wind_angle, "wind_to_direction", "degrees")
+        self.plugin = ResolveWindComponents()
+
+    def test_basic(self):
+        """Test function returns correct type"""
+        result = self.plugin.calc_true_north_offset(self.directions)
+        self.assertIsInstance(result, np.ndarray)
+
+    def test_values(self):
+        """Test that for UK National Grid coordinates the angle adjustments
+        are sensible"""
+        expected_result = np.array([
+            [3.012338, 2.635347, 2.258535, 1.881877, 1.505348],
+            [3.167531, 2.771182, 2.374997, 1.978951, 1.583021],
+            [3.310965, 2.896735, 2.482648, 2.068685, 1.654825]],
+            dtype=np.float32)
+        result = self.plugin.calc_true_north_offset(self.directions)
+        self.assertArrayAlmostEqual(RAD_TO_DEG*result, expected_result)
+
+
+class Test_resolve_wind_components(IrisTest):
+    """Tests the resolve_wind_components method"""
+
+    def setUp(self):
+        """Set up some arrays to convert"""
+        self.plugin = ResolveWindComponents()
+        wind_speed = 10.*np.ones((4, 4), dtype=np.float32)
+        wind_angle = np.array([[0., 30., 45., 60.],
+                               [90., 120., 135., 150.],
+                               [180., 210., 225., 240.],
+                               [270., 300., 315., 330.]])
+        self.wind_cube = set_up_cube(wind_speed, "wind_speed", "knots")
+        self.directions = set_up_cube(
+            wind_angle, "wind_to_direction", "degrees")
+        self.adjustments = np.zeros((4, 4), dtype=np.float32)
+
+    def test_basic(self):
+        """Test function returns correct type"""
+        uspeed, vspeed = self.plugin.resolve_wind_components(
+            self.wind_cube, self.directions, self.adjustments)
+        self.assertIsInstance(uspeed, iris.cube.Cube)
+        self.assertIsInstance(vspeed, iris.cube.Cube)
+
+    def test_values(self):
+        """Test correct values are returned for well-behaved angles"""
+        expected_uspeed = 5.*np.array(
+            [[0., 1., np.sqrt(2.), np.sqrt(3.)],
+             [2., np.sqrt(3.), np.sqrt(2.), 1.],
+             [0., -1., -np.sqrt(2.), -np.sqrt(3.)],
+             [-2., -np.sqrt(3.), -np.sqrt(2.), -1.]])
+
+        expected_vspeed = 5*np.array(
+            [[2., np.sqrt(3.), np.sqrt(2.), 1.],
+             [0., -1., -np.sqrt(2.), -np.sqrt(3.)],
+             [-2., -np.sqrt(3.), -np.sqrt(2.), -1.],
+             [0., 1., np.sqrt(2.), np.sqrt(3.)]])
```

Comment:
I think all the expected results are back-to-front. An Easterly wind (direction=90 degrees) should have a negative u component. A northerly wind (direction=0 degrees) should have a negative v component.

Reply:
As discussed - we're working with "wind_to_direction" at this point.